shardingsphere-operator/pkg/controllers/shardingsphere_chaos_controller.go

```diff
@@ -213,10 +258,183 @@ func (r *ShardingSphereChaosReconciler) reconcileStatus(ctx context.Context, nam
 	if err != nil {
 		return err
 	}
-	rt.Status = ssChaos.Status
+	setRtStatus(rt, ssChaos)
 	return r.Status().Update(ctx, rt)
 }
 
+func getRequirement(ssChaos *sschaosv1alpha1.ShardingSphereChaos) reconcile.InjectRequirement {
+	var jobName reconcile.InjectRequirement
+	if ssChaos.Status.Phase == sschaosv1alpha1.PhaseBeforeExperiment || ssChaos.Status.Phase == sschaosv1alpha1.PhaseAfterExperiment {
+		jobName = reconcile.Experimental
+	}
+	if ssChaos.Status.Phase == sschaosv1alpha1.PhaseInChaos {
+		jobName = reconcile.Pressure
+	}
+	if ssChaos.Status.Phase == sschaosv1alpha1.PhaseRecoveredChaos {
+		jobName = reconcile.Verify
+	}
+	return jobName
+}
+
+func getJobSuccessAndFail(conditions []batchV1.JobCondition) (bool, bool) {
+	var (
+		failed    = false
+		succeeded = false
+	)
+
+	for i := range conditions {
+		r := conditions[i]
+		if r.Type == batchV1.JobComplete {
+			if r.Status == v1.ConditionTrue {
+				succeeded = true
+			}
+			if r.Status == v1.ConditionFalse {
+				succeeded = false
+			}
+		}
+
+		if r.Type == batchV1.JobFailed {
+			if r.Status == v1.ConditionTrue {
+				failed = true
+			}
+			if r.Status == v1.ConditionFalse {
+				failed = false
+			}
+		}
+	}
+
+	return succeeded, failed
+}
+
+func setDefault(ssChaos *sschaosv1alpha1.ShardingSphereChaos) {
+	if ssChaos.Status.Phase == "" {
+		ssChaos.Status.Phase = sschaosv1alpha1.PhaseBeforeExperiment
+	}
+	if ssChaos.Status.Result == nil || len(ssChaos.Status.Result) == 0 {
```

Comment:
no need to determine `len(ssChaos.Status.Result) == 0`, coz it's equals `ssChaos.Status.Result = []sschaosv1alpha1.Result{}`